Python/Product/TestAdapter.Executor/UnitTest/TestDiscovererUnitTest.cs

```diff
@@ -26,62 +58,90 @@ internal class TestDiscovererUnitTest : IPythonTestDiscoverer {
-                        unitTestResults = JsonConvert.DeserializeObject<List<UnitTestDiscoveryResults>>(json);
-                    } catch (InvalidOperationException ex) {
-                        Error("Failed to parse: {0}".FormatInvariant(ex.Message));
-                        Error(json);
-                    } catch (JsonException ex) {
-                        Error("Failed to parse: {0}".FormatInvariant(ex.Message));
-                        Error(json);
+                    _settings.PathEnv,
+                    _settings.DiscoveryWaitTimeInSeconds
+                    );
+            } catch (TimeoutException) {
+                Error(Strings.PythonTestDiscovererTimeoutErrorMessage);
+                return;
+            }
+
+            List<UnitTestDiscoveryResults> results = null;
+            try {
+                results = JsonConvert.DeserializeObject<List<UnitTestDiscoveryResults>>(json);
+            } catch (InvalidOperationException ex) {
+                Error("Failed to parse: {0}".FormatInvariant(ex.Message));
+                Error(json);
+            } catch (JsonException ex) {
+                Error("Failed to parse: {0}".FormatInvariant(ex.Message));
+                Error(json);
+            }
+
+            CreateVsTests(results, logger, discoverySink);
+        }
+
+        private string ExecuteProcess(
+            string filename,
+            Dictionary<string, string> env,
+            IEnumerable<string> arguments,
+            string workingDirectory,
+            string pathEnv,
+            int timeoutInSeconds
+            ) {
+            using (var outputStream = new MemoryStream())
+            using (var reader = new StreamReader(outputStream, Encoding.UTF8, false, 4096, leaveOpen:true))
+            using (var writer = new StreamWriter(outputStream, Encoding.UTF8, 4096, leaveOpen: true))
+            using (var proc = ProcessOutput.Run(
+                filename,
+                arguments,
+                workingDirectory,
+                env,
+                visible: false,
+                new StreamRedirector(writer)
+            )) {
+                DebugInfo("cd " + workingDirectory);
+                DebugInfo("set " + pathEnv + "=" + env[pathEnv]);
+                DebugInfo(proc.Arguments);
+                
+                if (!proc.ExitCode.HasValue) {
+                    if (!proc.Wait(TimeSpan.FromSeconds(timeoutInSeconds))) {
+                        try {
+                            proc.Kill();
+                        } catch (InvalidOperationException) {
+                            // Process has already exited
+                        }
+                        throw new TimeoutException();
                     }
                 }
-            } catch (Exception ex) {
-                Error(ex.Message);
+                writer.Flush();
+                outputStream.Seek(0, SeekOrigin.Begin);
+                string data = reader.ReadToEnd();
+                return data;
```

Comment:
same comment as above applies here with regards to using `StandardOutputLines`
also, since the code is identical, maybe this method can go in a static class under Services folder.

Reply:
exactly what i was thinking. left that change out but can add it.